modules/social_features/social_group/modules/social_group_flexible_group/modules/social_flexible_group_book/social_flexible_group_book.module

```diff
@@ -94,14 +98,157 @@ function social_flexible_group_book_form_node_form_alter(array &$form, FormState
   // a specific global permission - "create new books".
   // Group managers with only "verified" global role should be able to create
   // new books as well even without this permission.
+  if (!isset($form['book']['bid']['#options'])) {
+    return;
+  }
   $nid = !$node->isNew() ? $node->id() : 'new';
   $options = $form['book']['bid']['#options'];
   if (!isset($options[$nid])) {
-    if ($nid == 'new' || ($nid != $node->book['original_bid'])) {
+    if ($nid === 'new' || ($nid !== $node->book['original_bid'])) {
       // The node can become a new book, if it is not one already.
       $form['book']['bid']['#options'] = [$nid => t('- Create a new book -')] + $options;
     }
   }
+
+  // Do not allow for user to move book if it's a parent because this may cause
+  // troubles with "Parent item" field in "Book outline" form element.
+  // Also, disable the "Group" field if the book is creating in group context.
```

Comment:
While looking at the tests I noticed this change. I'm not against it, but I do worry about the fact that we're introducing an inconsistency between the creation of a topic/event in a group and a book.

If I create a topic or event through "Create topic" or "Create book" then I can still change the group, whereas with a book this is now prevented.

Has thought been given to this consistency in UX? Is there a plan to make the same change for topics/events?

Reply:
Unfortunately, this issue cause a problem with displaying sub-books with the same parent but different groups...
More details can be found in the [initial](https://getopensocial.atlassian.net/browse/PROD-21759) ticket. 

As I know there are no any plans to do the same for topics/events.

Reply:
I saw the ticket, but there's two scenarios right?

>  // Do not allow for user to move book if it's a parent because this may cause
>  // troubles with "Parent item" field in "Book outline" form element.

This makes sense and avoids the issue mentioned.

> Also, disable the "Group" field if the book is creating in group context.
This is the UX change I'm talking about that I don't understand the reasoning for. Especially if we're not mirroring that to topic/events.